apps/pie-storybook/stories/testing/pie-switch.test.stories.ts

```diff
@@ -0,0 +1,238 @@
+import { html, nothing } from 'lit';
+import { action } from '@storybook/addon-actions';
+import { type Meta } from '@storybook/web-components';
+
+import '@justeattakeaway/pie-switch';
+import { type SwitchProps, labelPlacements, defaultProps } from '@justeattakeaway/pie-switch';
+import '@justeattakeaway/pie-icons-webc/dist/IconCheck.js';
+
+import { createStory, createVariantStory, type TemplateFunction } from '../../utilities';
+
+type SwitchStoryMeta = Meta<SwitchProps>;
+
+const defaultArgs: SwitchProps = {
+    ...defaultProps,
+    label: 'Label',
+    aria: {
+        label: 'switch label',
+        describedBy: 'switch description',
+    },
+    name: 'switch',
+    value: 'switchValue',
+};
+
+const switchStoryMeta: SwitchStoryMeta = {
+    title: 'Switch',
+    component: 'pie-switch',
+    argTypes: {
+        checked: {
+            description: 'Same as the HTML checked attribute - indicates whether the switch is on or off',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.checked,
+            },
+        },
+        disabled: {
+            description: 'Same as the HTML disabled attribute - indicates whether the switch is disabled or not',
+            control: 'boolean',
+            defaultValue: {
+                summary: defaultProps.disabled,
+            },
+        },
+        label: {
+            description: 'The label text for the switch',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: 'Label',
+                },
+            },
+        },
+        labelPlacement: {
+            description: 'Set the placement of the label.',
+            control: 'select',
+            if: { arg: 'label', truthy: true },
+            options: labelPlacements,
+            defaultValue: {
+                summary: defaultProps.labelPlacement,
+            },
+        },
+        aria: {
+            description: 'The ARIA labels used for the switch.',
+            control: 'object',
+        },
+        name: {
+            description: 'Same as the HTML name attribute - indicates the name of the switch (for use with forms)',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: 'switch',
+                },
+            },
+        },
+        value: {
+            description: 'Same as the HTML value attribute - indicates the value of the switch (for use with forms). Defaults to "on".',
+            control: {
+                type: 'text',
+                defaultValue: {
+                    summary: defaultProps.value,
+                },
+            },
+        },
+        required: {
+            description: 'Same as the HTML required attribute - for use in forms',
```

Comment:
As these are automated tests, I think we can remove descriptions and tidy up this file - I've left this comment on every other test migration PR too :)